runtime/src/debug/cpp/SourceInfo.cpp

```diff
@@ -71,7 +71,7 @@ CSSourceInfoRef (*CSSymbolOwnerGetSourceInfoWithAddress)(
 
 const char* (*CSSourceInfoGetPath)(CSSourceInfoRef info);
 
-uint32_t (*CSSourceInfoGetLineNumber)(CSSourceInfoRef info);
+int32_t (*CSSourceInfoGetLineNumber)(CSSourceInfoRef info);
```

Comment:
[Here](https://opensource.apple.com/source/IOKitTools/IOKitTools-97/ioclasscount.tproj/ioclasscount.c) it's used with `printf`'s `%d`, so it's probably signed.

Reply:
On the other hand: https://github.com/mozilla/positron/blob/4b0595d06faa1617bd989a8c47231576f30ea6b2/widget/cocoa/nsCocoaDebugUtils.h#L77

Reply:
A couple of experiments showed that for files larger than 3 billion lines of code, some debug info (notably, symbol location) isn't being read (or not even generated, never checked). For files with 2 billion lines of code, everything is fine.

Nevertheless, restored `uint32_t` here and added loud checks if it ever overflows `int32_t` boundaries.